compiler/optimizer/Inliner.cpp

```diff
@@ -4591,12 +4609,28 @@ void TR_InlinerBase::inlineFromGraph(TR_CallStack *prevCallStack, TR_CallTarget
 
             if (site)
                {
+#ifdef J9_PROJECT_SPECIFIC
+               TR_PrexArgInfo* callerArgInfo = calltarget->_prexArgInfo;
+               TR_PrexArgInfo* argInfo = new (trStackMemory()) TR_PrexArgInfo(node->getNumArguments(), trMemory());
+               TR_PrexArgInfo* argsFromCaller = TR_PrexArgInfo::argInfoFromCaller(node, argInfo, callerArgInfo);
+
                for (int32_t i = 0; i < site->numTargets(); i++)
                   {
                   TR_CallTarget *target = site->getTarget(i);
+                  // Need to propagate and validate the arg info here, non-invariant arg info should have been cleared by now
+                  // Arg info from caller
+                  target->_prexArgInfo = TR_PrexArgInfo::enhance(target->_prexArgInfo, argsFromCaller, comp());;
                   getUtil()->computePrexInfo(target);
                   targetsToInline.add(target);
                   }
+#else
+               for (int32_t i = 0; i < site->numTargets(); i++)
```

Comment:
I'm not sure this is the right factoring of this change. The `getUtil()->computePrexInfo(target)` was meant to isolate the project specific part of this loop down into the projects so that we didn't need `#ifdef` code in here. If there isn't enough info passed to `computePrexInfo` could we find a way to keep this prex part of this contained to the Util where the project specific extension can exist? Right now the logic is weirdly split and the duplicated loop is not ideal...

Reply:
Added an util that computes prex info with target and target's caller arg info. The j9 specific code has been moved down to openj9.